src/integration/java/io/pinecone/integration/controlPlane/pod/CreateDescribeListAndDeleteIndexTest.java

```diff
@@ -1,51 +1,181 @@
+    public static void cleanUp() {
+        // Delete the index
+        controlPlaneClient.deleteIndex(indexName);
+    }
+
+    @Test
+    public void describeAndListIndex() {
         // Describe the index
         IndexModel indexModel = controlPlaneClient.describeIndex(indexName);
         assertNotNull(indexModel);
         assertEquals(10, indexModel.getDimension());
         assertEquals(indexName, indexModel.getName());
         assertEquals(IndexMetric.COSINE, indexModel.getMetric());
+        assertNotNull(indexModel.getSpec().getPod());
+        assertEquals("p1.x1", indexModel.getSpec().getPod().getPodType());
 
         // List the index
         IndexList indexList = controlPlaneClient.listIndexes();
         assertNotNull(indexList.getIndexes());
+        assertTrue(indexList.getIndexes().stream().anyMatch(index -> indexName.equals(index.getName())));
+    }
 
-        // Delete the index
-        controlPlaneClient.deleteIndex(indexName);
-        Thread.sleep(3500);
+    @Test
+    public void createIndexWithPodsAndPodType() {
+        String podIndexName = RandomStringBuilder.build("create-pod", 8);
+        CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().environment(environment).pods(2).podType(
+                "p1.x2");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().pod(podSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name(podIndexName)
+                .metric(IndexMetric.COSINE)
+                .dimension(10)
+                .spec(createIndexRequestSpec);
+
+        IndexModel createdIndex = controlPlaneClient.createIndex(createIndexRequest);
+        assertEquals(createdIndex.getName(), podIndexName);
+        assertEquals(createdIndex.getSpec().getPod().getPods(), 2);
+        assertEquals(createdIndex.getSpec().getPod().getPodType(), "p1.x2");
+        assertEquals(createdIndex.getStatus().getReady(), false);
+        assertEquals(createdIndex.getStatus().getState(), IndexModelStatus.StateEnum.INITIALIZING);
+
+        controlPlaneClient.deleteIndex(podIndexName);
+    }
+
+    @Test
+    public void createIndexWithInvalidName() {
+        CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().environment(environment).podType("p1.x1");
+        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().pod(podSpec);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
+                .name("Invalid-name")
+                .metric(IndexMetric.COSINE)
+                .dimension(10)
+                .spec(createIndexRequestSpec);
+
+        try {
+            controlPlaneClient.createIndex(createIndexRequest);
+
+            fail("Expected to throw PineconeBadRequestException");
+        } catch (PineconeBadRequestException expected) {
+            assertTrue(expected.getLocalizedMessage().contains("must consist of lower case alphanumeric characters or" +
```

Comment:
Nit: you might want to change the error msg here to say *what* must consist of lowercase (one word) chars... so something like "Index name must consist..." or similar

Reply:
For the validations on error messages, I was trying to keep them somewhat minimal per this feedback: https://github.com/pinecone-io/pinecone-java-client/pull/81#discussion_r1527630404

So I'm using `contains` to validate that specific substrings are in the error message, which is why some of these look a little odd at first glance. My thinking was a minimal string may help prevent things failing if the error messages change, but I feel like that's somewhat unlikely and it may be beneficial to include more of the error in the assert.

I'll touch things up based on your feedback, thanks!